### DISCUSSION TITLE
Writing to MongoDB help please!

### DISCUSSION THREAD

**Discussion by KernOil** [Budibase/budibase#1315] [Help] [answered]
Hi @mike12345567  @joebudi @shogunpurple  guys, 

I could really use your help. I really enjoy your BB platform I think it is going to be huge! 

I can not get the data out of BB to save my life. I have tried the webhooks and could not get that to work I have tried the external data connection create for Mongo DB and I keep getting  Query Error: Cannot create property '_id' on string"
I am tying to build a data collection program for work but I cant get data out. would you please help me.  

Thank you very much,
Andrew Ramay

**Comment by czarnndz** [comment]
Hi, @KernOil Did you already figure out how to write to MongoDB from BB? I'm having the same error

**Comment by melohagan** [comment, marked as answer]
Hi @czarnndz 

Could you show me some screenshots please? What version of Budibase are you using?

There have been some updates/fixes to the connector since OP: https://github.com/Budibase/budibase/pull/5877

**Reply by czarnndz** [reply]
Hi @melohagan, thanks for reply.

I'm using BB 1.0.196 

What I'm trying to to is post a form to MongoDB

I setup a Query using bindings variables as query values

![image](https://user-images.githubusercontent.com/7936586/172873112-c369d9cd-716d-48cd-bc1d-0b0ea4ce470a.png)

But when I click "Run Query" I got below message:

`Query Error: Cannot create property '_id' on string '{ "name": Daniel Herrera, "profile_img": https://drive.google.com/uc?export=view&id=1i7jZp2vyUoYhL1G9jF7oXb09r6l7huLD, "region": Perú, "email": daniel97hh@gmail.com, "phone": +51 912908145, "nationality": Peruana, "experience_years": 8 meses de experiencia, "time_notice": 3 semanas, "salary_expectation": 1,700 USD Libres, "tech_profile": Jr. Backend RoR, "sales_pitch": Desarrollador Full Stack apasionado por la tecnología con 8 meses de experiencia desarrollando con Ruby On Rails, siempre buscando ampliar sus conocimientos y afrontar nuevos retos. Graduado del bootcamp de desarrollo web de Le Wagon. Le gusta estar constantemente actualizado en nuevas tecnologías y metodologías. Buscando profundizar sus conocimientos en desarrollo Backend preferentemente con Ruby, bases de datos, APIs y testing, para poder aportar valor a las organizaciones y crecer como desarrollador. }'`

**Reply by czarnndz** [reply]
@melohagan I just figure out the error, looks like binding replace exactly the values in query but not as string as I guessed, So it was misforming the json to mongodb, So what I did was set binding default values inside quotes ("") and now it's working. But idk if this is how it should work

![image](https://user-images.githubusercontent.com/7936586/172875304-1bfc9633-635a-4f98-9b12-f3ac32bfd4d4.png)

**Reply by melohagan** [reply]
Thanks for the screenshots! 
That is how it should work, you should really place your bindings within quotes rather than have the quotes in the binding value, e.g: https://docs.budibase.com/docs/mongodb#bindings

This is because you may want bindings to form part of one value, e.g. `"name": "{{firstname}} {{lastname}}"`

**Reply by czarnndz** [reply]
Okay I got it now, thank you for your help!!!
